pycloudlib/oci/cloud.py

```diff
@@ -366,7 +366,11 @@ def snapshot(self, instance, clean=True, name=None):
             current_data=image_data,
             desired_state="AVAILABLE",
         )
-
-        self.created_images.append(image_data.id)
+        
+        # only add image to created_images list for cleanup if keep is False
+        if not keep:
+            self.created_images.append(image_data.id)
+        else:
+            self._log.info("Keeping image %s", image_data.display_name)
```

Comment:
I think it's worth us persisting a list of `self.kept_images` or `self.retained_mages`. Then on teardown/`cloud.clean()`  we can make sure we emit a footer log or something conspicuous that says `f"Skipping deletion of preserved  image(s): {self.retained_images}"`. Otherwise there is little hope we will see this inline log during image creation during our busy integration tests.

Also we probably want to log both name and id so that we can reference that image in the future on CLI/SDK/cloud-dashboards.
```suggestion
            self._log.info("Keeping image %s[id:%s]", image_data.display_name, image_data.id)
```

Reply:
Agreed! Thanks for the feedback @blackboxsw!